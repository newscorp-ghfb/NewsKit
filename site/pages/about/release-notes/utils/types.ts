import {ReleaseType} from 'semver';

export interface Release {
  published_at: string;
  tag_name: string;
  body: string;
}

export interface FullRelease extends Release {
  change_level: ReleaseType;
}

export interface ReleasesPageProps {
  releases: Release[];
}

export type ReleaseBadge = ReleaseType | 'breaking change';
