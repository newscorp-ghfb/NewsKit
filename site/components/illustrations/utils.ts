import {SVGAttributes} from 'react';
import {Theme} from '../../../src/theme';

// @ts-ignore
export const toCSSVar = (theme: Theme, props) => {
  const inlineStyle = {} as SVGAttributes<SVGPathElement>;
  const newProps = {...props};

  if (theme.colors[props?.fill!]) {
    inlineStyle.fill = `var(--color-${props.fill})`;
    delete newProps.fill;
  }
  if (theme.colors[props?.stroke!]) {
    inlineStyle.stroke = `var(--color-${props.fill})`;
    delete newProps.stroke;
  }
  return {
    style: inlineStyle,
    ...newProps,
  };
};

const escapeRegExp = (string: string): string =>
  // $& means the whole matched string
  string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');

export const pathToID = (path: string) => {
  if (path) {
    return path
      .replace(new RegExp(escapeRegExp('../'), 'g'), '')
      .replace(new RegExp(escapeRegExp('./'), 'g'), '')
      .replace(new RegExp(escapeRegExp('/'), 'g'), '-');
  }
  return '';
};

export const isSafari =
  typeof window !== 'undefined' &&
  /^((?!chrome|android).)*safari/i.test(window.navigator.userAgent);
