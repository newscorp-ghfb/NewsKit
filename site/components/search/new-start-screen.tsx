import React from 'react';
import NextLink from 'next/link';
import {
  Cell,
  Grid,
  TextBlock,
  LinkStandalone,
  Stack,
  toNewsKitIcon,
} from 'newskit';
import {ChevronRight as FilledChevronRight} from '@emotion-icons/material/ChevronRight';

const IconFilledChevronRight = toNewsKitIcon(FilledChevronRight);

const startScreenOptions = [
  {
    category: {
      name: 'Getting started',
    },
    items: [
      {
        name: 'Design overview',
        href: '/getting-started/design/design-overview/',
      },
      {
        name: 'Engineering overview',
        href: '/getting-started/code/engineering-overview/',
      },
      {
        name: 'Instrumentation setup',
        href: '/getting-started/code/instrumentation/',
      },
    ],
  },
  {
    category: {
      name: 'Popular searches',
    },
    items: [
      {name: 'Components', href: '/components/overview/'},
      {name: 'Theme', href: '/theme/overview/'},
      {name: 'Button', href: '/components/button/'},
    ],
  },
  {
    category: {
      name: 'Presets',
    },
    items: [
      {
        name: 'Style presets',
        href: '/theme/presets/style-presets/',
      },
      {name: 'Transition presets', href: '/theme/presets/transition-presets/'},
      {
        name: 'Typography presets',
        href: '/theme/presets/typography-presets/',
      },
    ],
  },
  {
    category: {
      name: 'Foundations',
    },
    items: [
      {name: 'Design tokens', href: '/theme/foundation/design-tokens/'},
      {name: 'Colours', href: '/theme/foundation/colours/'},
      {name: 'Fonts', href: '/theme/foundation/fonts/'},
    ],
  },
];

export const NewStartScreen = () => (
  <div className="DocSearch-NewStartScreen">
    <Grid
      xsMargin="space030"
      smMargin="space030"
      mdMargin="space030"
      lgMargin="space030"
      xlMargin="space030"
    >
      {startScreenOptions.map(({category, items}) => (
        <Cell key={category.name} xs={6}>
          <TextBlock
            marginBlockStart="space010"
            marginBlockEnd="space050"
            stylePreset="inkContrast"
            typographyPreset="utilityHeading010"
          >
            {category.name}
          </TextBlock>
          <Stack
            spaceInline="space045"
            marginBlockStart="space010"
            marginBlockEnd="space080"
          >
            {items.map(({name, href}) => (
              <NextLink legacyBehavior href={href} passHref>
                <LinkStandalone
                  key={name}
                  href={href}
                  overrides={{typographyPreset: 'utilityLink020'}}
                >
                  {name}
                  <IconFilledChevronRight overrides={{size: 'iconSize010'}} />
                </LinkStandalone>
              </NextLink>
            ))}
          </Stack>
        </Cell>
      ))}
    </Grid>
  </div>
);
