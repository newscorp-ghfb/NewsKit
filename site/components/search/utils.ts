import {DocSearchHit} from '@docsearch/react/dist/esm/types';

export const ignoreFilter = (item: DocSearchHit): boolean => {
  // this includes components overview from results
  if (item.url.includes('components/overview')) {
    return false;
  }

  /*
    Remove introduction result 
    because these 2 results are the same:
    .../components/radio-button/
    .../components/radio-button/#introduction
  */
  if (item.url.endsWith('#introduction')) {
    return false;
  }

  // other filters in the future

  return true;
};

export const mapPathname = (item: DocSearchHit): DocSearchHitExtended => {
  const parseUrl = document.createElement('a');
  parseUrl.href = item.url;

  const canonicalAs = parseUrl.pathname + parseUrl.hash;
  const canonicalPathname = canonicalAs
    .replace(/^\/api/, '/api-docs')
    .replace(/#(.*)$/, '')
    .replace(/\/$/, '');

  return {
    ...item,
    as: canonicalAs,
    pathname: canonicalPathname,
  };
};
