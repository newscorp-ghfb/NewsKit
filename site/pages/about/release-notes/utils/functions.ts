import {diff} from 'semver';
import {FullRelease, Release} from './types';
import {GITHUB_API_URL, REPO} from './constants';

export async function fetchGitHubReleases(per_page: number = 10) {
  const res = await fetch(
    `${GITHUB_API_URL}/repos/${REPO}/releases?per_page=${per_page}`,
  );
  const data = await res.json();
  return data;
}

const sortReleases = (ascending: boolean) => (
  {tag_name: a}: Release,
  {tag_name: b}: Release,
) =>
  // eslint-disable-next-line no-nested-ternary
  (a > b ? 1 : b > a ? -1 : 0) * (ascending ? 1 : -1);

// This function takes an array of Releases and calculates their change_levels
// by comparing against the previous release in the array. It returns an array
// containing all releases (with change levels) in the original array except the
// oldest (this cannot be calculated as there is no previous release to compare
// it against, so it is not included in the output array).
export const addChangeLevelToReleases: (
  releases: Release[],
) => FullRelease[] = releases => {
  const ascending = releases.sort(sortReleases(true));
  return ascending
    .slice(1) // The oldest release is not included in the output.
    .reduce(
      (prev: FullRelease[], next: Release) => [
        ...prev,
        {
          ...next,
          // Compare against the previous release in the array or the oldest
          // release removed above.
          change_level:
            next.tag_name === 'v5.4.2' // todo: remove this (only here for design review)
              ? 'major'
              : diff((prev[0] || ascending[0]).tag_name, next.tag_name)!, // Only null if the version is the same (never).
        },
      ],
      [],
    )
    .sort(sortReleases(false));
};
