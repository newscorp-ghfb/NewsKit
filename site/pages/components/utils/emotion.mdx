import Layout from '../../../components/layout';

export default Layout;

# Emotion

## Overview
NewsKit exports out some of the most commonly used Emotion functions/types/components to help consumers build custom components where needed. 

## CSS
The css function in Emotion accepts styles as a template literal, object, or array of objects and returns a class name. It is the foundation of emotion.
The primary way to style elements with emotion is the `css` prop. It provides a concise and flexible API to style your components.

## Styled Components
`styled` is a way to create React components that have styles attached to them. It’s available from @emotion/styled. 
`styled` is very similar to `css` except you call it with an HTML tag or React component and then call that with a template literal for string styles or a regular function call for object styles.
NewsKit uses `styled` in both the components and the documentation site. Here is the official Emotion documentation for [Styled Components](https://emotion.sh/docs/styled).

## CSSObject
The `css` prop accepts object styles (`CSSObject`) directly and does not require an additional import. Object styles (`CSSObject`) are especially useful with the css prop because you don’t need a css call like with string styles.
Object styles can also be used with `styled`. Here is the official Emotion documentation for [Object Styles](https://emotion.sh/docs/css-prop)

## SerializedStyles
To pass string styles(`SerializedStyles`), you must use css which is exported by @emotion/core, it can be used as a tagged template literal.
Here is the official Emotion documentation for [String Styles](https://emotion.sh/docs/css-prop).

## Global
Global component (`Global`) injects styles into the global scope and is useful for applications such as css resets or font faces.
It accepts a styles prop which accepts the same values as the css prop except it inserts styles globally. Global styles are also removed when the styles change or when the Global component unmounts.
In NewsKit, we use `Global` to apply font faces to our documentation site pages. Here is the official Emotion documentation for [Global Styles](https://emotion.sh/docs/globals).

## Theming
Theming is included in the @emotion/react package. Emotion's theming API provide us with `ThemeProvider`，`withTheme` and `useTheme`, which are widely used in NewsKit components and the documentation site. 
You can read about their usage [here](/theme/theming/using-a-theme) or on Emotion's official documentation site [Theming](https://emotion.sh/docs/theming#api).