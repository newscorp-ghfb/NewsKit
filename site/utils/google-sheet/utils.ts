import {CMSData, CMSProps, CMSResponse} from './types';

export class CMSError extends Error {
  constructor(msg: string) {
    super(msg);
    this.name = 'CMSError';
  }
}

export const formatSheetData = <T extends CMSProps>(data: CMSData): T => {
  const transformedArray = data.map(([key, value]) => [
    key.toLowerCase(),
    value,
  ]);
  // Hack needed to avoid JSON-Serialization validation error from Next.js https://github.com/zeit/next.js/discussions/11209
  // >>> Reason: `undefined` cannot be serialized as JSON. Please use `null` or omit this value all together.
  return JSON.parse(JSON.stringify(Object.fromEntries(transformedArray)));
};

export const parseCMSResponse = <T extends CMSProps>(
  cmsResponse: CMSResponse,
  {
    required,
    optional = {},
    dynamic = {},
  }: {
    required: Record<string, string>;
    optional?: Record<string, string>;
    dynamic?: Record<string, string>;
  },
): T => {
  if (!cmsResponse) {
    throw new CMSError('No CMS data found');
  }
  const content = formatSheetData<T>(cmsResponse);

  const missingKeys = Object.keys(required).filter(key => !content[key]);
  const invalidKeys = Object.keys(content).filter(key => {
    if (required[key] || optional[key]) {
      return false;
    }
    const prefixMatch = Object.keys(dynamic).find(k =>
      key.match(`${k}\\d+\\b`),
    );
    return !prefixMatch;
  });

  let err = ``;
  if (missingKeys.length) {
    err = `MISSING_KEYS: ${missingKeys.join(', ')}. `;
  }
  if (invalidKeys.length) {
    err = `${err}INVALID_KEYS: ${invalidKeys.join(', ')}. `;
  }

  if (err.length) {
    throw new CMSError(err);
  }

  return content;
};

export const getCMSPropsWithPrefix = <T extends Record<string, string>>(
  content: CMSProps,
  prefix: keyof T & string,
) => Object.entries(content).filter(([k]) => k.startsWith(prefix));
