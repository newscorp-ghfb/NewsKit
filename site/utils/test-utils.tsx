/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme: any = renderToFragmentWithThemeFactory(
  docsThemeLight,
);

export const mockNewsKitComponents = (...components: string[]) => () => ({
  ...(jest.requireActual('newskit') as any),
  ...components.reduce((acc, comp) => {
    acc[comp] = ({children, ...props}: any) => (
      <div
        data-comp={`NewsKit ${comp}`}
        data-testid={comp}
        data-props={JSON.stringify(props, null, 2)}
      >
        {children}
      </div>
    );
    return acc;
  }, {} as Record<string, Function>),
});
