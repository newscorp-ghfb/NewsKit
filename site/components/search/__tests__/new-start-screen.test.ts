import {renderToFragmentWithTheme} from '../../../utils/test-utils';
import {NewStartScreen} from '../new-start-screen';

describe('NewStartScreen', () => {
  test('renders correctly', () => {
    const fragment = renderToFragmentWithTheme(NewStartScreen);
    expect(fragment).toMatchSnapshot();
  });
});
