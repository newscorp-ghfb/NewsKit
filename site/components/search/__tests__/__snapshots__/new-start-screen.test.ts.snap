// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewStartScreen renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 100%;
  max-width: 1920px;
  margin: 0 auto;
}

.emotion-1 {
  box-sizing: border-box;
  -webkit-background-clip: padding-box;
  background-clip: padding-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

@media screen {
  .emotion-1 {
    margin: -16px 4px 0 4px;
  }
}

@media screen and (min-width: 480px) {
  .emotion-1 {
    margin: -16px 4px 0 4px;
  }
}

@media screen and (min-width: 768px) {
  .emotion-1 {
    margin: -16px 4px 0 4px;
  }
}

@media screen and (min-width: 1024px) {
  .emotion-1 {
    margin: -24px 0px 0 0px;
  }
}

@media screen and (min-width: 1440px) {
  .emotion-1 {
    margin: -24px 0px 0 0px;
  }
}

.emotion-2 {
  box-sizing: border-box;
  -webkit-background-clip: padding-box;
  background-clip: padding-box;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

@media screen {
  .emotion-2 {
    padding: 0 8px;
    margin-top: 16px;
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    max-width: 50%;
  }
}

@media screen and (min-width: 1024px) {
  .emotion-2 {
    padding: 0 12px;
    margin-top: 24px;
  }
}

.emotion-3 {
  margin: 0;
  color: #09111C;
  font-family: "Poppins",sans-serif;
  font-size: 16px;
  line-height: 18px;
  font-weight: 700;
  letter-spacing: 0;
  padding: 0.5px 0px;
  margin-block-start: 4px;
  margin-block-end: 24px;
}

.emotion-3 svg {
  fill: #09111C;
}

.emotion-3::before {
  content: '';
  margin-bottom: -0.2155em;
  display: block;
}

.emotion-3::after {
  content: '';
  margin-top: -0.2125em;
  display: block;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  margin-top: calc(-20px/2);
  margin-bottom: calc(-20px/2);
  margin-block-start: 4px;
  margin-block-end: 48px;
}

.emotion-5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-top: calc(20px/2);
  margin-bottom: calc(20px/2);
}

.emotion-6 {
  display: inline-block;
  color: #3358CC;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-6 {
    transition-property: color,fill;
    transition-duration: 200ms,200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1),cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-6 {
    transition-property: color,fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1),cubic-bezier(0, 0, .5, 1);
  }
}

.emotion-6 svg {
  fill: #3358CC;
}

.emotion-6:hover:not(:disabled) {
  color: #254CAC;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-6:hover:not(:disabled) svg {
  fill: #254CAC;
}

.emotion-6:active:not(:disabled) {
  color: #12387A;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-6:active:not(:disabled) svg {
  fill: #12387A;
}

.emotion-6:focus-visible:not(:disabled) {
  outline-color: #3768FB;
  outline-style: solid;
  outline-width: 2px;
  outline-offset: 2px;
}

@media not all and (min-resolution: 0.001dpcm) {
  @supports (-webkit-appearance: none) and (stroke-color: transparent) {
    .emotion-6:focus-visible:not(:disabled) {
      outline-style: auto;
    }
  }
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  margin-left: calc(-4px/2);
  margin-right: calc(-4px/2);
}

.emotion-8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: calc(4px/2);
  margin-right: calc(4px/2);
}

.emotion-9 {
  margin: 0;
  font-family: "Poppins",sans-serif;
  font-size: 14px;
  line-height: 21px;
  font-weight: 400;
  letter-spacing: 0;
  padding: 0.5px 0px;
  display: inline-block;
  display: block;
}

.emotion-9::before {
  content: '';
  margin-bottom: -0.403em;
  display: block;
}

.emotion-9::after {
  content: '';
  margin-top: -0.4em;
  display: block;
}

.emotion-11 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  fill: #2E3F54;
  vertical-align: unset;
  display: inline-block;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-11 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-11 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

.emotion-11.emotion-11 {
  width: 16px;
  height: 16px;
}

<div
    class="DocSearch-NewStartScreen"
  >
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1"
      >
        <div
          class="emotion-2"
        >
          <p
            class="emotion-3"
          >
            Getting started
          </p>
          <div
            class="emotion-4"
          >
            <div
              class="emotion-5"
            >
              <a
                class="emotion-6"
                href="/getting-started/design/design-overview/"
              >
                <span
                  class="emotion-7"
                >
                  <span
                    class="emotion-8"
                  >
                    <span
                      class="emotion-9"
                    >
                      Design overview
                    </span>
                  </span>
                  <span
                    class="emotion-8"
                  >
                    <svg
                      aria-hidden="true"
                      class="emotion-11 emotion-12"
                      fill="currentColor"
                      focusable="false"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </span>
                </span>
              </a>
            </div>
            <div
              class="emotion-5"
            >
              <a
                class="emotion-6"
                href="/getting-started/code/engineering-overview/"
              >
                <span
                  class="emotion-7"
                >
                  <span
                    class="emotion-8"
                  >
                    <span
                      class="emotion-9"
                    >
                      Engineering overview
                    </span>
                  </span>
                  <span
                    class="emotion-8"
                  >
                    <svg
                      aria-hidden="true"
                      class="emotion-11 emotion-12"
                      fill="currentColor"
                      focusable="false"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </span>
                </span>
              </a>
            </div>
            <div
              class="emotion-5"
            >
              <a
                class="emotion-6"
                href="/getting-started/code/instrumentation/"
              >
                <span
                  class="emotion-7"
                >
                  <span
                    class="emotion-8"
                  >
                    <span
                      class="emotion-9"
                    >
                      Instrumentation setup
                    </span>
                  </span>
                  <span
                    class="emotion-8"
                  >
                    <svg
                      aria-hidden="true"
                      class="emotion-11 emotion-12"
                      fill="currentColor"
                      focusable="false"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </span>
                </span>
              </a>
            </div>
          </div>
        </div>
        <div
          class="emotion-2"
        >
          <p
            class="emotion-3"
          >
            Popular searches
          </p>
          <div
            class="emotion-4"
          >
            <div
              class="emotion-5"
            >
              <a
                class="emotion-6"
                href="/components/overview/"
              >
                <span
                  class="emotion-7"
                >
                  <span
                    class="emotion-8"
                  >
                    <span
                      class="emotion-9"
                    >
                      Components
                    </span>
                  </span>
                  <span
                    class="emotion-8"
                  >
                    <svg
                      aria-hidden="true"
                      class="emotion-11 emotion-12"
                      fill="currentColor"
                      focusable="false"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </span>
                </span>
              </a>
            </div>
            <div
              class="emotion-5"
            >
              <a
                class="emotion-6"
                href="/theme/overview/"
              >
                <span
                  class="emotion-7"
                >
                  <span
                    class="emotion-8"
                  >
                    <span
                      class="emotion-9"
                    >
                      Theme
                    </span>
                  </span>
                  <span
                    class="emotion-8"
                  >
                    <svg
                      aria-hidden="true"
                      class="emotion-11 emotion-12"
                      fill="currentColor"
                      focusable="false"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </span>
                </span>
              </a>
            </div>
            <div
              class="emotion-5"
            >
              <a
                class="emotion-6"
                href="/components/button/"
              >
                <span
                  class="emotion-7"
                >
                  <span
                    class="emotion-8"
                  >
                    <span
                      class="emotion-9"
                    >
                      Button
                    </span>
                  </span>
                  <span
                    class="emotion-8"
                  >
                    <svg
                      aria-hidden="true"
                      class="emotion-11 emotion-12"
                      fill="currentColor"
                      focusable="false"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </span>
                </span>
              </a>
            </div>
          </div>
        </div>
        <div
          class="emotion-2"
        >
          <p
            class="emotion-3"
          >
            Presets
          </p>
          <div
            class="emotion-4"
          >
            <div
              class="emotion-5"
            >
              <a
                class="emotion-6"
                href="/theme/presets/style-presets/"
              >
                <span
                  class="emotion-7"
                >
                  <span
                    class="emotion-8"
                  >
                    <span
                      class="emotion-9"
                    >
                      Style presets
                    </span>
                  </span>
                  <span
                    class="emotion-8"
                  >
                    <svg
                      aria-hidden="true"
                      class="emotion-11 emotion-12"
                      fill="currentColor"
                      focusable="false"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </span>
                </span>
              </a>
            </div>
            <div
              class="emotion-5"
            >
              <a
                class="emotion-6"
                href="/theme/presets/transition-presets/"
              >
                <span
                  class="emotion-7"
                >
                  <span
                    class="emotion-8"
                  >
                    <span
                      class="emotion-9"
                    >
                      Transition presets
                    </span>
                  </span>
                  <span
                    class="emotion-8"
                  >
                    <svg
                      aria-hidden="true"
                      class="emotion-11 emotion-12"
                      fill="currentColor"
                      focusable="false"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </span>
                </span>
              </a>
            </div>
            <div
              class="emotion-5"
            >
              <a
                class="emotion-6"
                href="/theme/presets/typography-presets/"
              >
                <span
                  class="emotion-7"
                >
                  <span
                    class="emotion-8"
                  >
                    <span
                      class="emotion-9"
                    >
                      Typography presets
                    </span>
                  </span>
                  <span
                    class="emotion-8"
                  >
                    <svg
                      aria-hidden="true"
                      class="emotion-11 emotion-12"
                      fill="currentColor"
                      focusable="false"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </span>
                </span>
              </a>
            </div>
          </div>
        </div>
        <div
          class="emotion-2"
        >
          <p
            class="emotion-3"
          >
            Foundations
          </p>
          <div
            class="emotion-4"
          >
            <div
              class="emotion-5"
            >
              <a
                class="emotion-6"
                href="/theme/foundation/design-tokens/"
              >
                <span
                  class="emotion-7"
                >
                  <span
                    class="emotion-8"
                  >
                    <span
                      class="emotion-9"
                    >
                      Design tokens
                    </span>
                  </span>
                  <span
                    class="emotion-8"
                  >
                    <svg
                      aria-hidden="true"
                      class="emotion-11 emotion-12"
                      fill="currentColor"
                      focusable="false"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </span>
                </span>
              </a>
            </div>
            <div
              class="emotion-5"
            >
              <a
                class="emotion-6"
                href="/theme/foundation/colours/"
              >
                <span
                  class="emotion-7"
                >
                  <span
                    class="emotion-8"
                  >
                    <span
                      class="emotion-9"
                    >
                      Colours
                    </span>
                  </span>
                  <span
                    class="emotion-8"
                  >
                    <svg
                      aria-hidden="true"
                      class="emotion-11 emotion-12"
                      fill="currentColor"
                      focusable="false"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </span>
                </span>
              </a>
            </div>
            <div
              class="emotion-5"
            >
              <a
                class="emotion-6"
                href="/theme/foundation/fonts/"
              >
                <span
                  class="emotion-7"
                >
                  <span
                    class="emotion-8"
                  >
                    <span
                      class="emotion-9"
                    >
                      Fonts
                    </span>
                  </span>
                  <span
                    class="emotion-8"
                  >
                    <svg
                      aria-hidden="true"
                      class="emotion-11 emotion-12"
                      fill="currentColor"
                      focusable="false"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </span>
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
